### ISSUE TITLE
Please add some keyboard shortcuts to the search UI

### ISSUE THREAD

**Issue by Schnouki** [Wilfred/deadgrep#59]
In search results, it would be *super* useful (for me) to have several shortcuts to change search parameters. For instance:

- `t`: change search term
- `y s`, `y w`, `y r`: change search type to strings, words, regexp
- `C s`, `C e`, `C i`: set case to smart, sensitive, ignore
- `c n`, `c b`, `c a`: change context to none, before, after
- `d`: change directory
- `f a`, `f t`, `f g`: change files to all, type, glob

In other words, please let me change all search parameters without having to move the point to a button and then hitting `RET`! :)

**Comment by Gei0r** [comment]
Additional idea: Have a key to toggle trough the different options in the current line.

**Comment by s-kostyaev** [comment]
And also  will be great to add some keys for adding `-u` and `-uu` arguments.

**Comment by bobslee** [comment]
Just to help here (FYI).
Following prompts the user for a new file type, then restarts the search.

```elisp
(defun deadgrep-file-type ()
  "Prompt the user for a new file type, then restart the search."
  (interactive)
  (let ((new-file-type
         (deadgrep--read-file-type deadgrep--initial-filename)))
    (setq deadgrep--file-type (cons 'type new-file-type)))
  (rename-buffer
   (deadgrep--buffer-name deadgrep--search-term default-directory) t)
  (deadgrep-restart))
```

A `use-package` key binding:

```elisp
:bind (:map deadgrep-mode-map
            ("T" . deadgrep-file-type))
```

**Comment by Schnouki** [comment]
```elisp
    (define-key map (kbd "S") #'deadgrep-search-term)
    (define-key map (kbd "T") #'deadgrep-cycle-search-type)
    (define-key map (kbd "C") #'deadgrep-cycle-search-case)
    (define-key map (kbd "F") #'deadgrep-cycle-files)
    (define-key map (kbd "D") #'deadgrep-directory)
    (define-key map (kbd "^") #'deadgrep-parent-directory)
```

Amazing, that certainly covers my needs. Thanks!
